File: portfolio-backend/src/br/com/wilton/portfolio/dao/WorkDoneDao.java
package br.com.wilton.portfolio.dao;

import java.util.List;

import javax.ejb.Stateless;
import javax.persistence.TypedQuery;

import br.com.wilton.portfolio.model.WorkDone;

@Stateless
public class WorkDoneDao extends GenericDao<WorkDone> {

	@Override
	protected Class<WorkDone> getClassType() {
		return WorkDone.class;
	}

	public List<WorkDone> findByProfile(long id) {
		TypedQuery<WorkDone> query = em.createQuery("select w from work_done w inner join w.profile p "
				+ "where p.idProfile = :idProfile ", WorkDone.class);
		query.setParameter("idProfile", id);
		return query.getResultList();
	}

}

File: portfolio-backend-node/db/WorkDoneDao.js
class WorkDoneDao {
    constructor(connection) {
        this._connection = connection;
    }

    findByProfileId(id, callback) {
        this._connection.query(`select * from work_done 
          where profile_idProfile = ?`,[id], callback);
    }
}

module.exports = () => {
    return WorkDoneDao;
}